createdisk.sh

```diff
@@ -104,25 +104,24 @@ EOF
 
 # Add gvisor-tap-vsock service
 ${SSH} core@${VM_IP} 'sudo bash -x -s' <<EOF
-  tee /etc/systemd/system/gv-user-network@.service <<TEE
+  tee /etc/systemd/system/gvisor-tap-vsock.service <<TEE
 [Unit]
 Description=gvisor-tap-vsock Network Traffic Forwarder
 After=NetworkManager.service
-BindsTo=sys-devices-virtual-net-%i.device
-After=sys-devices-virtual-net-%i.device
 
 [Service]
 Restart=on-failure
 Environment="GV_VSOCK_PORT=1024"
 EnvironmentFile=-/etc/sysconfig/gv-user-network
-ExecStart=/usr/libexec/podman/gvforwarder -preexisting -iface %i -url vsock://2:"\\\${GV_VSOCK_PORT}"/connect
+ExecStartPre=/bin/sh -c 'for i in {1..10}; do ip link show tap0 && exit 0; sleep 1; done; exit 1'
+ExecStart=/usr/libexec/podman/gvforwarder -preexisting -iface tap0 -url vsock://2:"\\\${GV_VSOCK_PORT}"/connect
```

Comment:
I thought podman was using a unit file similar to ours, but what they have is a NM configuration file for `tap0` (they call it `vsock0`) which is not autostarted, and then in `ExecStartPost` (so after starting `gvforwarder`), they explicitly call `nmcli conn up vsock0` https://github.com/containers/podman/blob/main/pkg/machine/hyperv/stubber.go#L523-L550

Reply:
In our case we do need to make sure it is autostarted because during our execution we make change to resolve.conf after running `ovs-configuration` and restart NM so if it is not autostarted then in our case tap0 fails to start.